elf/io/extensions.py

```diff
@@ -114,34 +114,46 @@ def noop(*args, **kwargs):
 
 
 try:
-    # will not override z5py
     import zarr
-    if int(zarr.__version__.split('.', maxsplit=1)[0]) > 2:
-        raise ImportError(
-            f"zarr {zarr.__version__} (>=3.0.0) is not supported. "
-            "Please downgrade to zarr 2 if you require zarr support. "
-            "See https://github.com/constantinpape/elf/issues/114 "
-            "for more details."
-        )
-
-    # zarr stores cannot be used as context managers,
-    # which breaks compatibility with similar libraries.
-    # This wrapper patches in those methods.
-    @functools.wraps(zarr.open)
-    def zarr_open(*args, **kwargs):
-        """@private
-        """
-        z = zarr.open(*args, **kwargs)
-        ztype = type(z)
-        if not hasattr(ztype, "__enter__"):
-            ztype.__enter__ = identity
-        if not hasattr(ztype, "__exit__"):
-            ztype.__exit__ = noop
-        return z
-
-    register_filetype(
-        zarr_open, N5_EXTS + ZARR_EXTS, zarr.hierarchy.Group, zarr.core.Array, True
-    )
+
+    # Check which zarr version is available.
+    try:
+        zarr_major_version = int(zarr.__version__.split(".", maxsplit=1)[0])
+    exception Exception:
```

Comment:
```suggestion
    except Exception:
```

my suggestion needs an exception....

Reply:
:D Good point :)